extension/src/Dictionary.ts

```diff
@@ -0,0 +1,165 @@
+import { existsSync, readFileSync, writeFileSync } from "fs";
+
+export class Dictionary implements IDictonary {
+  language: string;
+  wordList: DictPair[];
+  dictionaryFile: string;
+
+  constructor(dictionaryFile: string) {
+    this.dictionaryFile = dictionaryFile;
+    if (!existsSync(dictionaryFile)) {
+      throw new Error(`Could not open file: "${this.dictionaryFile}"`);
+    }
+    const d = JSON.parse(
+      readFileSync(this.dictionaryFile, "utf8")
+    ) as IDictonary;
+    this.language = d.language;
+    this.wordList = d.wordList;
+  }
+
+  toJSON(): unknown {
+    return {
+      language: this.language,
+      wordList: this.wordList,
+    };
+  }
+
+  saveDictionary(): void {
+    const spaces = 2;
+    writeFileSync(
+      this.dictionaryFile,
+      JSON.stringify(this.toJSON(), null, spaces),
+      "utf8"
+    );
+  }
+
+  addWord(word: string, replaceWith: string): Dictionary {
```

Comment:
Agreed, `Dictionary` maybe isn't the most useful. Going with `DictPair` mostly to enable method chaining. It could prove useful :) If you don't like it we'll do `void` :)

Reply:
I'll vote for `DictPair` as well